edx/edx-enterprise
# Manage Learners

**Description:** This PR adds learner management view to enterprise Django admin.

**JIRA:** [ENT-48](https://openedx.atlassian.net/browse/ENT-48)

**Dependencies:** cross-dependent on https://github.com/edx/edx-platform/pull/14003

**Installation instructions:**
1. Switch to this branch.
2. Make sure edx-enterprise is installed into edx-platform virtualenv locally (i.e. `pip install -e /edx/src/edx-enterprise` or `pip install -e git+https://...`)
3. Add `'django_object_actions'` to  `OPTIONAL_APPS` in common.py.

**Testing instructions:**
1. Open Enterprise Customer record in Django admin
2. In the top-right corner, next to "History" button, "Manage Learners" button should appear. Click on it.
3. You will arrive to "Manage Learners" view for that particular Enterprise Customer.
4. Check that breadcrumbs strip is correct (should contain link to Enterprise Customer record and link to lit of ECs)
5. Try linking a learner via form on the right side
5.1. By username - if user exists, his email should be added to a list. If not, validation error should be shown
5.2. By email - if invalid email is put, validation message should be shown. If email of an existing user is provided, username is shown in the list
5.3. In both cases, if user is already linked to EC (including current), validation error should be shown.
6. Make sure added learners appear in the list.
6.1. Make sure they survive page refresh.
6.2. Make sure they appear in Enterprise Customer User admin with correct data (i.e. linked to correct EC)
7. Try unlinking users
7.1. Make sure confirmation is shown.
7.2. Make sure survives page refresh.
7.3. Make sure they disappear from Enterprise Customer User admin.

**Reviewers:**
- [x] @saleem-latif and/or @asadiqbal08 

**Merge checklist:**
- [x] Submit updates to cookiecutter template (docs, consistent `'` vs. `"` in setup.py etc.)

**Post merge:**

**Author concerns:**
1. Lack of tests for JS part
2. Might be a good time to add integration tests - but this would be time-consuming.
